server/libs/domain/src/partner/partner.core.ts

```diff
@@ -0,0 +1,38 @@
+import { BadRequestException, InternalServerErrorException, Logger } from '@nestjs/common';
+import { IPartnerRepository, PartnerDirection } from './partner.repository';
+import { PartnerEntity } from '@app/infra/entities';
+
+export class PartnerCore {
+  readonly logger = new Logger(PartnerCore.name);
+
+  constructor(private repository: IPartnerRepository) {}
+
+  getAll(userId: string, direction: PartnerDirection): Promise<PartnerEntity[]> {
+    return this.repository.getAll(userId, direction);
+  }
+
+  get(sharedBy: string, sharedWith: string): Promise<PartnerEntity | null> {
+    return this.repository.get(sharedBy, sharedWith);
+  }
+
+  create(sharedBy: string, sharedWith: string): Promise<PartnerEntity> {
+    try {
+      return this.repository.create({
+        sharedBy: sharedBy,
+        sharedWith: sharedWith,
+      });
+    } catch (error: any) {
+      this.logger.error(error, error.stack);
+      throw new InternalServerErrorException('failed to create partner');
+    }
+  }
```

Comment:
I don't know if I care about catching and rethrowing an internal exception here or not. We don't really do it most places. I might prefer not re-throwing, but just letting the original error bubble and handle it somewhere else.

Reply:
Removed the same rethrowing in ShareCore also